packages/core/src/types/Matrix2D.ts

```diff
@@ -0,0 +1,663 @@
+import {EPSILON, PossibleMatrix2D} from './Matrix';
+import {PossibleVector2, Vector2} from './Vector';
+
+/**
+ * A specialized 2x3 Matrix representing a 2D transformation.
+ *
+ * A Matrix2D contains six elements defined as
+ * [a, b,
+ *  c, d,
+ *  tx, ty]
+ *
+ * This is a shortcut for a 3x3 matrix of the form
+ * [a, b, 0,
+ *  c, d, 0
+ *  tx, ty, 1]
+ *
+ * Note that because a Matrix2D ignores the z-values of each component vectors,
+ * it does not satisfy all properties of a "real" 3x3 matrix.
+ *
+ *   - A Matrix2D has no transpose
+ *   - A(B + C) = AB + AC does not hold for a Matrix2D
+ *   - (rA)^-1 = r^-1 A^-1, r != 0 does not hold for a Matrix2D
+ *   - r(AB) = (rA)B = A(rB) does not hold for a Matrix2D
+ */
+export class Matrix2D {
```

Comment:
Could we also add some getters for retrieving the x and y unit vectors from a matrix?

Reply:
So the equivalent of calling `.column(0)` and `.column(1)`?

Reply:
Nevermind, I forgot it's a 3x2 matrix and `column` already returns a Vector2